File: Node/第六章/09-ai-interview-coach/server/src/scripts/doctor.ts
import { PostgresSaver } from '@langchain/langgraph-checkpoint-postgres'
import { DEFAULT_POSTGRES_URI } from '../storage/storage.service.js'

try {
	process.loadEnvFile()
} catch {
	// Replay 模式可以不创建 .env。
}

const uri = process.env.POSTGRES_URI ?? DEFAULT_POSTGRES_URI
const checkpointer = PostgresSaver.fromConnString(uri)

try {
	await checkpointer.setup()
	console.log('✓ PostgreSQL 连接正常')
	console.log(
		process.env.DEEPSEEK_API_KEY
			? `✓ AI 模式可用：${process.env.DEEPSEEK_MODEL ?? 'deepseek-v4-flash'}`
			: '· 未配置 DEEPSEEK_API_KEY，将使用 Replay 模式'
	)
} finally {
	await checkpointer.end()
}

File: Python/第六章/09-ai-interview-coach/doctor.py
"""检查本地项目运行依赖。"""

from __future__ import annotations

import os

from storage import create_storage


def main() -> None:
    storage = create_storage()

    try:
        storage.setup()
        print("✓ PostgreSQL 连接正常")
        print(
            f"✓ AI 模式可用：{os.getenv('DEEPSEEK_MODEL', 'deepseek-v4-flash')}"
            if os.getenv("DEEPSEEK_API_KEY")
            else "· 未配置 DEEPSEEK_API_KEY，将使用 Replay 模式"
        )
    finally:
        storage.close()


if __name__ == "__main__":
    main()
